src/bot.coffee

```diff
@@ -188,14 +207,28 @@ class SlackBot extends Adapter
           @receive new TopicMessage user, event.topic, event.ts
         when undefined
           @robot.logger.debug "Received message in channel: #{channel.name || channel.id}, from: #{user.name}"
-          @receive new SlackTextMessage(user, undefined, undefined, event, channel, @robot.name)
-        # NOTE: if we want to expose all remaining subtypes not covered above as a generic message implement an else
-        # else
-        #   # other subtypes may not have user or user.room defined
+          
+          SlackTextMessage.makeSlackTextMessage(user || bot, undefined, undefined, event, channel, @robot.name, @client, (message) =>
+            @receive message
+          )
+          .catch((error) =>
+            @robot.logger.error "Error constructing SlackTextMessage"
+          )
+        else
+          @receive new CatchAllMessage {user}
+
     else if event.type is 'reaction_added' or event.type is 'reaction_removed'
+      return unless event.user && event.item_user
+
+      # If the reaction is from a stranger or on a stranger's message, then the event will be ignored
+      # Strangers have no shared channel in common, external members have shared channel membership
+      # External member reactions should function normally
+      return if event.item_user.is_stranger || event.user.is_stranger
```

Comment:
this definitely reflects what [the docs](https://api.slack.com/shared-channels#users_may_seem_strange) say, but i think this is worth double checking in an actual workspace with shared channels. from my memory trying it out, this doesn't seem right.